src/android/StatusBar.java

```diff
@@ -206,6 +212,16 @@ public void run() {
         return false;
     }
 
+    private float getStatusBarHeight() {
+        float statusBarHeight = 0;
+        int resourceId = cordova.getActivity().getResources().getIdentifier("status_bar_height", "dimen", "android");
+        if (resourceId > 0) {
+            float scaleRatio = cordova.getActivity().getResources().getDisplayMetrics().density;
+            statusBarHeight = cordova.getActivity().getResources().getDimension(resourceId) / scaleRatio;
+        }
+        return statusBarHeight;
```

Comment:
Tested on a v27 emulator and on a v29 device, it works; Thanks!

I think that this feature will be used a lot to expose the height in a CSS variable, and I think that it might make sense to expose it directly in the plugin. I'm currently setting the CSS variable like this:
```js
const onSuccess = (height: number): void => {
  const roundedHeight = Math.ceil(height);
  if (isNaN(roundedHeight)) {
    document.documentElement.style.setProperty('--cordova-status-bar-height', '0px');
    console.warn('Failed to determine the StatusBar height.')
  } else {
    document.documentElement.style.setProperty('--cordova-status-bar-height', `${height}px`);
  }
};
```
And in CSS:
```css
.my-toolbar-container {
    padding-top: calc(8px + var(--cordova-status-bar-height));
}
```
But in order for this to work properly in all cases, the CSS variable value needs to be updated on each screen orientationchange. Because, in some cases, the StatusBar height depends on the height of the screen notch and if you rotate the screen, then the status bar height might be different depending if you have a notch or not.

It would be nice if the user could just set a:
```
<preference name="StatusBarHeightCSSVariable" value="true" />
```
In order to use a CSS variable directly, without having to write any JS in order to fetch the height and to subscribe to the screen orientation changes.

Reply:
Perhaps https://developer.mozilla.org/en-US/docs/Web/API/Window/orientationchange_event could help to detect the screen orientation change and recompute the statusbar height?

Reply:
That works, but if the CSS variable is defined, you don't have to care about anything. You just use it as it is. 

This is the same approach that the `iOS` guys took in order to avoid issues with screen notches. The've added those `env(safe-area-inset-*)` custom CSS "environment variables".

* https://developer.mozilla.org/en-US/docs/Web/CSS/env

And they get updated when you rotate the screen and the screen notch is on a different screen edge.

Reply:
I know, I use the Safari `safe-area-inset-*` variables since iPhone X. What you are suggesting is an Android Chrome bugfix plugin. Basically, a Corodva plugin for Android without an exposed JavaScript interface, that sets the `safe-area-inset-*` env variables in the Android WebView's document. The plugin could also handle orientation changes. I think the Statusbar plugin is not the correct place to fix the missing env variables of Chrome.

Reply:
It's not a Chrome Bugfix, because that standard doesn't exist yet, it's still a draft, and from what I can tell, you can't provide those `env()` variables in a polyfill.

And I'm also not talking about implementing the `env(safe-area-inset-*)` as part of the StatusBar, I'm just saying that the status bar should always expose it's own height.

And now that I think about it, the CSS variable should have a fixed value and it should not depend on orientation changes.